access-control-service-sql-impl/src/main/java/org/eclipse/tractusx/semantics/accesscontrol/sql/service/SqlBackedAccessControlRuleService.java

```diff
@@ -76,15 +78,24 @@ public List<String> filterValidSpecificAssetIdsForLookup(
    @Override
    public ShellVisibilityCriteria fetchVisibilityCriteriaForShell( ShellVisibilityContext shellContext, String bpn ) throws DenyAccessException {
       Set<AccessRulePolicy> matchingAccessControlRules = findMatchingAccessControlRules( shellContext, bpn );
-      Set<String> visibleSpecificAssetIdNames = matchingAccessControlRules.stream()
-            .flatMap( accessControlRule -> accessControlRule.getVisibleSpecificAssetIdNames().stream() )
-            .collect( Collectors.toSet() );
+
+      Map<String, Set<String>> visibleSpecificAssetIds = new HashMap<>();
+      matchingAccessControlRules.forEach( rule -> {
+         rule.getMandatorySpecificAssetIds().forEach( specificAssetId -> {
+            visibleSpecificAssetIds.computeIfAbsent( specificAssetId.name(), k -> new HashSet<>() ).add( specificAssetId.value() );
+         } );
+
+         rule.getVisibleSpecificAssetIdNames().forEach( specificAssetIdName -> {
```

Comment:
Hi,
Can it cause an issue that we are including both the specific and the any match in this collection? I assume not, because the evaluation will depend on the exact `specificAssetId` values of the shell in the end, but wanted to raise it just in case I am wrong...

Reply:
Done

Reply:
Documenting the potential edge case we have discussed: If the mandatory Id is matching the shell based on "A" property, but the visible contains only "B", then a potential duplication with the "B" name would become visible. Need to confirm with Thomas what we want exactly.